### PR TITLE
Enhanced RDoc for Symbol

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats (I got carried away):
- #==
- #inspect
- #name
- #to_s
- #to_sym
- #to_proc (sort of)
- #succ
- #<=>
- #casecmp
- #casecmp?
- #=~
- #match
- #match?

### CODE REVIEW

File: string.c [hunk 1]
```diff
@@ -11475,12 +11479,13 @@ rb_sym_proc_call(ID mid, int argc, const VALUE *argv, int kw_splat, VALUE passed
 
 #if 0
 /*
- * call-seq:
- *   sym.to_proc
+ *  call-seq:
+ *    to_proc
+ *
+ *  Returns a Proc object corresponding to +self+:
```
Comment (peterzhu2118) [inline string.c:11485]:
```suggestion
 *  Returns a Proc object which calls the method with name of +self+ on the first parameter and passes the remaining parameters to the method.
```

File: string.c [hunk 2]
```diff
@@ -11475,12 +11479,13 @@ rb_sym_proc_call(ID mid, int argc, const VALUE *argv, int kw_splat, VALUE passed
...
  *
- * Returns a _Proc_ object which responds to the given method by _sym_.
+ *    :foo.to_proc # => #<Proc:0x000001afe0e48680(&:foo) (lambda)>
```
Comment (peterzhu2118) [inline string.c:11487]: Maybe we should also have an example usage:

```suggestion
 *    proc = :to_s.to_proc # => #<Proc:0x000001afe0e48680(&:to_s) (lambda)>
 *    proc.call(1000) # => "1000"
 *    proc.call(1000, 16) # => "3e8"
```

File: string.c [hunk 3]
```diff
@@ -11475,12 +11479,13 @@ rb_sym_proc_call(ID mid, int argc, const VALUE *argv, int kw_splat, VALUE passed
...
  *
- *   (1..3).collect(&:to_s)  #=> ["1", "2", "3"]
```
Comment (peterzhu2118) [inline string.c:11483]: I think it's worth mentioning the `&:` syntax.

File: string.c [hunk 4]
```diff
@@ -11528,27 +11542,9 @@ sym_cmp(VALUE sym, VALUE other)
...
  *
- *  Case-insensitive version of #<=>:
- *
- *    :aBcDeF.casecmp(:abcde)   # => 1
- *    :aBcDeF.casecmp(:abcdef)  # => 0
- *    :aBcDeF.casecmp(:abcdefg) # => -1
- *    :abcdef.casecmp(:ABCDEF)  # => 0
- *
- *  Returns +nil+ if the two symbols have incompatible encodings,
- *  or if +other_symbol+ is not a symbol:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply string.c:11541]: Done.

File: string.c (hunk 1 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 2 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 3 above)
Comment (BurdetteLamar) [reply string.c:11483]: Done.

File: string.c [hunk 5]
```diff
@@ -11482,9 +11482,13 @@ rb_sym_proc_call(ID mid, int argc, const VALUE *argv, int kw_splat, VALUE passed
...
  *
- *  Returns a Proc object corresponding to +self+:
+ *  Returns a Proc object which calls the method with name of +self+
+ *  on the first parameter and passes the remaining parameters to the method.
  *
- *    :foo.to_proc # => #<Proc:0x000001afe0e48680(&:foo) (lambda)>
+ *    proc = :to_s.to_proc # => #<Proc:0x000001afe0e48680(&:to_s) (lambda)>
+ *    proc.call(1000) # => "1000"
+ *    proc.call(1000, 16) # => "3e8"
+ *    (1..3).collect(&:to_s)  #=> ["1", "2", "3"] *
```
Comment (peterzhu2118) [inline string.c:11491]: I think there's a typo here:

```suggestion
 *    (1..3).collect(&:to_s)  # => ["1", "2", "3"]
```

File: string.c [hunk 6]
```diff
@@ -11563,28 +11563,9 @@ sym_casecmp(VALUE sym, VALUE other)
...
- *  Unicode case folding, +false+ if they are not equal:
- *
- *    :aBcDeF.casecmp?(:abcde)                  # => false
- *    :aBcDeF.casecmp?(:abcdef)                 # => true
- *    :aBcDeF.casecmp?(:abcdefg)                # => false
- *    :abcdef.casecmp?(:ABCDEF)                 # => true
- *    :"\u{e4 f6 fc}".casecmp?(:"\u{c4 d6 dc}") #=> true
- *
- *  Returns +nil+ if the two symbols have incompatible encodings,
- *  or if +other_symbol+ is not a symbol:
```
Comment (peterzhu2118) [inline string.c:11578]: This part should also be included just like `String#casecmp`.

File: string.c (hunk 5 above)
Comment (BurdetteLamar) [reply string.c]: Fixed.

File: string.c (hunk 6 above)
Comment (BurdetteLamar) [reply string.c:11578]: Done.

**Review by peterzhu2118 — APPROVED** [review]
Looks good

**Pull request merged**